bothan-core/src/tasks.rs

```diff
@@ -29,32 +29,29 @@ pub(crate) mod utils;
 #[derive(Debug, Clone, PartialEq)]
 pub struct Tasks {
     source_tasks: Vec<SourceTask>,
-    batched_signal_tasks: Vec<Vec<SignalTask>>,
+    batched_signal_tasks: Vec<SignalTask>,
 }
 
 impl Tasks {
```

Comment:
I recommend to rename Tasks to Job (my gut I prefer struct type name to be singular and reserve plural type to be vector of type ) Kor opinion noi @sorawit

Reply:
I prefer singular word for struct name too, but I think introducing new term "job" can make thing confusing. Maybe `TaskGroup`, `TaskBundle`, `TaskSet` or sth that signify it's a collection of tasks but singular?

Reply:
changed to `TaskSet`